cpg_utils/hail_batch.py

```diff
@@ -181,8 +165,37 @@ def full_path(self, prefix: str, suffix: str) -> str:
         return os.path.join(f'{self.scheme}://', prefix, suffix)
 
 
+class Namespace(Enum):
+    """
+    Storage namespace.
+    https://github.com/populationgenomics/team-docs/tree/main/storage_policies#main-vs-test
+    """
+
+    MAIN = 'main'
+    TEST = 'test'
+
+    @staticmethod
+    def from_access_level(str_val: str) -> 'Namespace':
+        """
+        Parse value from a access level string.
+        >>> Namespace.from_access_level('test')
+        Namespace.TEST
+        >>> Namespace.from_access_level('standard')
+        Namespace.MAIN
+        >>> Namespace.from_access_level('main')
+        Namespace.MAIN
+        """
+        for val, str_vals in {
+            Namespace.MAIN: ['main', 'standard', 'full'],
+            Namespace.TEST: ['test'],
+        }.items():
+            if str_val in str_vals:
+                return val
+        raise ValueError(f'Cannot parse namespace or access level {str_val}')
+
+
 def dataset_path(
-    suffix: str,
+    suffix: str = '',
```

Comment:
Is it useful to have an empty default value here? (Also below.)

Reply:
Sometimes it's useful to get the prefix, cast it to Path, and pass it around as a typed Path object, e.g. https://github.com/populationgenomics/production-pipelines/blob/initial/scripts/make_seqr_metadata.py#L62

Reply:
Not necessary for image_path() though, I'll remove

Reply:
What I tried to imply is that this is probably not the _default_ use case, hence it shouldn't be the default value. You can always get the same behavior by explicitly passing the empty string.